### DISCUSSION TITLE
why non-contiguous vector<bool>

### DISCUSSION THREAD

**Discussion by PaltryProgrammer** [microsoft/STL#3671] [General] [answered]
greetings kind regards may i please inquire why non-contiguous vector<bool> was chosen ? it seems to me if `vector<int>` is tolerated and accepted why not `vector<bool>` ? does not non-contiguous not limit programmer options ? perhaps even slow performance in some scenarios ? kind regards

**Comment by StephanTLavavej** [comment, marked as answer]
I have updated your post to add backticks, to prevent Markdown from eating angle brackets as HTML tags.

This was a notorious design decision in C++98; as I understand it, the partial specialization feature was new, and applying a space optimization to `vector<bool>` (making it store 8x less space) seemed like a good idea at the time. Its impact on generic code (and later concerns like multithreading) wasn't appreciated, and now it's too difficult to unwind (there have been attempts in the C++ Standardization Committee that have failed).
